### PR TITLE
[bdd-engine] Use empty params for substep parsing if there is no running scenarios

### LINKED ISSUE #1459: Conditional step crushes on the Before lifecycle
**Description:**
Conditional step is failing in Before story section, passes successfully in the Scenario section.

**STR:**
```
Description: test #{eval(...)} in Before story lifecycle

Meta:
    @toRun

Lifecycle:
Before:
Scope: STORY
When the condition `#{eval( "abc" == "abc" )}` is true I do
|step                                                            |
|When I initialize the SCENARIO variable `var` with value `value`|


Scenario: test scenario
When I initialize the SCENARIO variable `var` with value `value`
```

**Log and report attached:**
[log4.log](https://github.com/vividus-framework/vividus/files/6049187/log4.log)
[allure.zip](https://github.com/vividus-framework/vividus/files/6049212/allure.zip)

### PR DESCRIPTION
**Pull request by uarlouski**
Fixes #1459

### CODE REVIEW

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java [hunk 1]
```diff
@@ -53,8 +56,12 @@ public SubStepsConverter(ExtendedConfiguration configuration, IBddRunContext bdd
                         .collect(collectingAndThen(Collectors.toList(), DelegatingStoryReporter::new));
             }
             ExamplesTable subStepsTable = configuration.examplesTableFactory().createExamplesTable(subSteps);
-            List<Step> steps = stepExamplesTableParser.parse(subStepsTable,
-                    bddRunContext.getRunningStory().getRunningScenario().getExample());
+
+            Map<String, String> parameters = Optional.ofNullable(bddRunContext.getRunningStory().getRunningScenario())
+                                                     .map(RunningScenario::getExample)
+                                                     .orElse(Map.of());
```
Comment (ikalinin1) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java:62]: orElseGet(Map::of)

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java [hunk 2]
```diff
@@ -53,8 +56,12 @@ public SubStepsConverter(ExtendedConfiguration configuration, IBddRunContext bdd
                         .collect(collectingAndThen(Collectors.toList(), DelegatingStoryReporter::new));
             }
             ExamplesTable subStepsTable = configuration.examplesTableFactory().createExamplesTable(subSteps);
-            List<Step> steps = stepExamplesTableParser.parse(subStepsTable,
-                    bddRunContext.getRunningStory().getRunningScenario().getExample());
+
+            Map<String, String> parameters = Optional.ofNullable(bddRunContext.getRunningStory().getRunningScenario())
```
Comment (ikalinin1) [inline vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java:60]: Optional.ofNullable(bddRunContext.getRunnignStory()).map(..) ?

**Review by ikalinin1 — APPROVED** [review]

File: vividus-tests/src/main/resources/story/integration/LifecycleBeforeStorySubSteps.story [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Meta:
+    @epic vividus-core
+    @feature sub-steps
+    @issueId 1459
+
+Lifecycle:
+Before:
+Scope: STORY
+When the condition `true` is true I do
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/story/integration/LifecycleBeforeStorySubSteps.story:9]: Was it working with the given stories?

File: vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java (hunk 2 above)
Comment (uarlouski) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java:60]: bddRunContext.getRunnignStory() - can it be null?

Comment (ikalinin1) [reply vividus-bdd-engine/src/main/java/org/vividus/bdd/converter/SubStepsConverter.java:60]: You never know :)

File: vividus-tests/src/main/resources/story/integration/LifecycleBeforeStorySubSteps.story (hunk 3 above)
Comment (uarlouski) [reply vividus-tests/src/main/resources/story/integration/LifecycleBeforeStorySubSteps.story:9]: checked, yes

**Review by valfirst — APPROVED** [review]

**Pull request merged**